### PR TITLE
Add style guide for comments

### PR DESCRIPTION
**Pull request by jieyu**
Require that all comments to be full sentences.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -15,6 +15,13 @@ This also applies to the code snippets in the markdown files.
 
 * Please wrap the code at 72 characters.
 
+## Comments
```
Comment (akutz) [inline CONTRIBUTING.md]: Hi @jieyu,

I think the style of comments in Makefiles vs Source files vs etc. can be different. As long as the intent is understood and a basic style is adhered (line length, wrapping, etc), I would relax a little.

**Comment by akutz** [comment]
Hi @jieyu,

It's good practice to always include a useful description when filing a pull request. If none is available I recommend using the body of the PR's commit since [it is good practice](https://robots.thoughtbot.com/5-useful-tips-for-a-better-commit-message) for all Git commits to include a subject and body.

File: CONTRIBUTING.md (hunk 1 above)
Comment (jieyu) [inline CONTRIBUTING.md]: @akutz Comments are for human to read, thus making it a real sentence makes sense to me. I prefer we have consistency among all types of files.

Comment (akutz) [inline CONTRIBUTING.md]: > for human 

_for humans_

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -15,6 +15,13 @@ This also applies to the code snippets in the markdown files.
...
+
+This also applies to the code snippets in the markdown files.
+
+* End each sentence within a comment with a punctuation mark (please note that we generally prefer periods); this applies to incomplete sentences as well.
+* For trailing comments, leave one space.
```
Comment (jdef) [inline CONTRIBUTING.md]: > leave one space

... where? between the end of the code and the beginning of the comment? if so .. we should say that

Comment (jieyu) [inline CONTRIBUTING.md]: fixed.

**Pull request merged**